subprojects/file-temp/src/main/java/org/gradle/api/internal/file/DefaultTemporaryFileProvider.java

```diff
@@ -44,20 +46,20 @@ public File createTemporaryFile(String prefix, @Nullable String suffix, String..
         File dir = new File(baseDirFactory.create(), CollectionUtils.join("/", path));
         GFileUtils.mkdirs(dir);
         try {
-            return File.createTempFile(prefix, suffix, dir);
+            return TempFiles.createTempFile(prefix, suffix, dir);
         } catch (IOException e) {
             throw new UncheckedIOException(e);
         }
     }
 
     @Override
-    public File createTemporaryDirectory(@Nullable String prefix, @Nullable String suffix, @Nullable String... path) {
+    public File createTemporaryDirectory(@Nullable @Nonnull String prefix, @Nullable String suffix, String... path) {
         File dir = new File(baseDirFactory.create(), CollectionUtils.join("/", path));
         GFileUtils.mkdirs(dir);
         try {
             // TODO: This is not a great paradigm for creating a temporary directory.
             // See http://guava-libraries.googlecode.com/svn/tags/release08/javadoc/com/google/common/io/Files.html#createTempDir%28%29 for an alternative.
-            File tmpDir = File.createTempFile("gradle", "projectDir", dir);
+            File tmpDir = TempFiles.createTempFile(prefix, suffix, dir);
             tmpDir.delete();
             tmpDir.mkdir();
             return tmpDir;
```

Comment:
Why are we not using `Files.createTempDirectory` here?

Reply:
Same issue as the other one, needs to be used in the workers. This code needs more work before it's done, but the overarching goal of this PR was to try to preserve logical equivalence, and make the functional changes in a follow up.